### DISCUSSION TITLE
crazyflie not able to connect with cfclient

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1052] [Q&A] [answered]
I want to inquire about an issue I encountered when powering on the 'crazyflie' and attempting to connect it with 'cfclient' via 'crazyradio'. Unfortunately, I am unable to establish a connection, and the results displayed on the 'cfclient' console are as follows. Perhaps you can assist me in resolving this:

![Screenshot from 2023-11-24 16-35-58](https://github.com/bitcraze/discussions/assets/79006087/7ecede3f-295c-4cd0-8b9c-0012123df0c3)

I also attempted to connect the 'crazyflie' via USB, but unfortunately, I was unable to establish a connection through USB as well. When I ran the following command, this was the result:

Command: `sudo lsusb -v -d 0483:5740 | grep iSerial`
Result: `iSerial 3 560039001350465636393520`

Your assistance in resolving this matter would be greatly appreciated.

**Comment by ataffanel** [comment, marked as answer]
It does seems that the radio is working and is configured on the default channel 80. Does the Crazyflie start properly and passes self-test (all 4 motors should spin briefly making noise and the M4 LED should blink green a couple of time on startup). Then, when the client is communicating, the M4 LED should blink for each packet transferred. If that is not the case, then you have a problem with the firmware not starting properly.

**Reply by amitx64** [reply]
The M4 LED is not blinking. So, which firmware do I need to update?

**Reply by ataffanel** [reply]
The STM32 is controlling M4, so in your case it seems that the nRF51 is working properly but not the STM32, you should check that everything is working well in the STM32.

**Reply by amitx64** [reply]
@ataffanel, I've identified the issue. It turns out that I mistakenly updated my `crazyflie 2.1` with the `crazyflie 2.0` firmware. The console indicated a `MPU9250` status failure. However, upon reviewing the schematic for `crazyflie 2.1`, I discovered it actually uses `BMI088` and `BMI388`.

Upon cloning the `crazyflie-firmware` repository and attempting to build, I noticed that it was generating firmware for `crazyflie 2.0`. Upon inspecting the `makefile`, I couldn't find the `make command` to build firmware for `crazyflie 2.1`.

Could you please guide me on how to build firmware for `crazyflie 2.1`?

**Reply by ataffanel** [reply]
There is no building the firmware for Crazyflie 2.1: in factory the stm32 is prepared so that the Crazyflie firmware recognized a Crazyflie 2.1 platform. Since you are using your own clone Crazyflie, you can check the firmware platform handling to see how this is done. This mechanism was implemented to be able to support multiple platform from the same firmware binary.

**Reply by amitx64** [reply]
@knmcguire & @ataffanel, I followed firmware platform handling for `Crazyflie 2.1` and was able to connect with `CFClient` via `USB`, but I am unable to connect via `Crazyradio 2.0`. Here are the results:
![Screenshot from 2023-11-30 16-29-10](https://github.com/bitcraze/discussions/assets/79006087/6163aa4b-1b6a-4125-a869-87ba8ed78eba)
![Screenshot from 2023-11-28 16-37-49](https://github.com/bitcraze/discussions/assets/79006087/6d4c6cbb-1f43-435c-b2b8-cfb27d3ef76c)
![Screenshot from 2023-11-30 16-07-34](https://github.com/bitcraze/discussions/assets/79006087/7ab8a146-e887-40ba-9b5a-e98ba39b02eb)
![Screenshot from 2023-11-24 16-35-58](https://github.com/bitcraze/discussions/assets/79006087/512ca660-c78e-418e-9c58-80d1d9594ba3)
![Screenshot from 2023-11-30 17-04-05](https://github.com/bitcraze/discussions/assets/79006087/beb944c1-a9c6-4feb-b123-07651d79c866)

When I executed `python3 -m cfloader -w radio://0/80/2M flash cf21.bin stm32-fw`, my `Crazyflie 2.1` entered bootloader mode successfully, and both `LED_BLUE_L` of STM32F4 and `LED_BLUE` of NRF51 started blinking. However, I encountered an issue while attempting to flash the firmware, as depicted in the image below:
![Screenshot from 2023-12-01 12-23-55](https://github.com/bitcraze/discussions/assets/79006087/58f65fb1-9f7f-4dfa-b8bc-fd57b62b1586)

**Reply by amitx64** [reply]
@ataffanel, as evident from the results, I am unable to connect with `CFClient` via `Crazyradio 2.0`. Please guide me on resolving this issue.

**Reply by ataffanel** [reply]
There are many things that could be wrong here:
 - It does not seem that radio packets makes it to the STM32, the nRF51 does seems to get the radio packet though. Does the STM32 link led blinks when you try to connect? There could be a communication problem between the two that you need to debug.
 - Same thing for the bootloader, you need to make sure both bootloader actually starts and that data can make it between the two.

Again it is really hard for me to know what is happening with curstom hardware. My advice is to instrument your board as much as possible with debugger on both CPU and test points/leads to probe various signals with an oscilloscope.

**Reply by amitx64** [reply]
Thank you, @ataffanel! I have successfully connected the `Crazyflie 2.1` with `CFClient` using `Crazyradio 2.0`. Here is the results:

![connected_with_crazyflie_radio](https://github.com/bitcraze/discussions/assets/79006087/aed76043-ab6f-4626-8aa3-39ba819016bb)
